src/server/TCGAadapter.js

```diff
@@ -125,22 +177,33 @@ export class TCGAAdapter {
   }
 
   correlations = (table, gene, min, max) => {
+    const conditions = {
+      gene: gene
+    }
+    const projection = {
+      rvalue: 1,
+      _id: 0
+    }
     // Returns RBP names with corresponing Rvalue for gene in sorted order
-    return this.pool.query(`SELECT rvalue FROM ${table} WHERE gene = '${gene}'`).then(results => {
-      if (results.rows.length < 1) {
-        return [] // gene not found
-      }
-      // Pass just json with gene:correlation
-      const filteredRvals = rangeFilter(results.rows[0].rvalue, min, max)
-      // Map to object with gene and rvals as keys
-      const mappedRvals = _.map(filteredRvals, (value, gene) => {
-        return { gene: gene, Rvalue: value }
-      })
-      return _.reverse(
-        _.sortBy(mappedRvals, o => {
-          return o.Rvalue // We use reciprocal to achieve descending sort
+    return this.mongodb
```

Comment:
There's a lot of processing in this endpoint that we do outside of mongodb (`rangeFilter`, `mappedRvals`).

Is it possible to move that to mongodb using `aggregate`?

Reply:
Yes